### PR TITLE
Set up CI to verify and test each pull request

### PR DESCRIPTION
**Pull request by hzxuzhonghu**

### CODE REVIEW

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -0,0 +1,53 @@
+name: CI Lint
```
Comment (hzxuzhonghu) [inline .github/workflows/ci.yaml:1]: cc @zirain

File: .github/workflows/ci.yaml [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    runs-on: ubuntu-18.04
+    steps:
+      - name: checkout code
+        uses: actions/checkout@v2
+      - name: install Go
+        uses: actions/setup-go@v2
+        with:
+          go-version: 1.17.x
+      - name: make test
+        run: make test
```
Comment (gy95) [inline .github/workflows/ci.yaml:53]: add a new line

File: hack/golangci-lint.sh [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+util::install_tools ${GOLANGCI_LINT_PKG} ${GOLANGCI_LINT_VER}
+
+if golangci-lint run; then
+  echo 'Congratulations!  All Go source files have passed staticcheck.'
+else
+  echo # print one empty line, separate from warning messages.
+  echo 'Please review the above warnings.'
+  echo 'If the above warnings do not make sense, feel free to file an issue.'
+  exit 1
+fi
```
Comment (gy95) [inline hack/golangci-lint.sh:23]: add a new line

File: Makefile [hunk 4]
```diff
@@ -49,6 +51,9 @@ fix-copyright:
 	@${FINDFILES} \( -name '*.go' -o -name '*.cc' -o -name '*.h' -o -name '*.proto' -o -name '*.py' -o -name '*.sh' \) \( ! \( -name '*.gen.go' -o -name '*.pb.go' -o -name '*_pb2.py' \) \) -print0 |\
 		${XARGS} hack/fix_copyright_banner.sh
 
+lint:
+	hack/golangci-lint.sh
```
Comment (zirain) [inline Makefile:55]: rename to `golangci-lint`

File: Makefile [hunk 5]
```diff
@@ -40,6 +40,8 @@ kurator: clean
 		-o $(OUT_PATH)/kurator \
 		cmd/kurator/main.go
 
+.PHONY: verify
+verify: lint lint-copyright
```
Comment (zirain) [inline Makefile:44]: rename to `lint`

File: Makefile [hunk 6]
```diff
@@ -40,6 +40,8 @@ kurator: clean
 		-o $(OUT_PATH)/kurator \
 		cmd/kurator/main.go
 
+.PHONY: verify
+lint: lint lint-copyright
```
Comment (gy95) [inline Makefile:44]: should be `verify: lint lint-copyright`?

File: .github/workflows/ci.yaml [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+    steps:
+      - name: Install Go
+        uses: actions/setup-go@v2
+        with:
+          go-version: 1.17.x
+      - name: Checkout code
+        uses: actions/checkout@v2
+      - name: Run lint
+        run: |
+          make lint
```
Comment (gy95) [inline .github/workflows/ci.yaml:24]: make verify?

**Comment by kevin-wangzefeng** [comment]
one more try:
/approve

**Pull request merged**